cmd/server_test.go

```diff
@@ -1843,6 +1843,8 @@ func (s *TestSuiteCommon) TestGetPartialObjectMisAligned(c *C) {
 	1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,
 	1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,1234567890,
 	1234567890,1234567890,1234567890,123`
+	// ensuring that the buffer contains atleast 13 bytes of data.
+	buffer.WriteString(line[:13])
```

Comment:
This function is incredibly complex for what is being done here. The requirement is just to check that the requested range of an object returns the correct range of data.

The approach used to generate `putContent` (the string that is ultimately uploaded and parts of which are retrieved) is just obfuscated and has no method to it - this is why there is a weird crash and a weird special patch to handle it.

Here is a simple way to generate `putContent` that has better random properties:

1. Select a number of random number between 12 and (say) 100 randomly. Call it `n`. This is the length of the string.
2. Append "123456890" to an empty string slice `s` repeatedly until its length exceeds `n`. Then set `s = s[:n]`.
3. Run the [Knuth shuffle (also called Fisher-Yates shuffle)](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle) on `s`. The shuffle algo can be a simple utility function available for reuse in other places too.

That's it - use `s` as `putContent`.

As it stands the current code is unreadable and unmaintainable.